src/products/products.controller.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  Body,
+  Controller,
+  Get,
+  Logger,
+  Post,
+  Query,
+  UseGuards,
+} from '@nestjs/common';
+import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { AuthGuard } from '../auth/auth.guard';
+import { JwtProcessorType } from '../auth/auth.service';
+import { JwtType } from '../auth/jwt/jwt.type.decorator';
+import { CreateProductRequest } from './api/CreateProductRequest';
+import { ProductDto } from './api/ProductDto';
+import { ProductsService } from './products.service';
+
+@Controller('/api/products')
+@ApiTags('products controller')
+export class ProductsController {
+  private readonly logger = new Logger(ProductsController.name);
+
+  constructor(private readonly productsService: ProductsService) {}
+
+  @UseGuards(AuthGuard)
+  @JwtType(JwtProcessorType.RSA)
+  @Get()
+  @ApiOperation({
+    description: 'returns all products',
+  })
+  @ApiResponse({
+    type: ProductDto,
+    isArray: true,
+    status: 200,
+  })
+  async getProducts(): Promise<ProductDto[]> {
+    this.logger.debug('Get all products.');
+    const allProducts = await this.productsService.findAll();
+    return allProducts.map<ProductDto>(ProductDto.covertToApi);
```

Comment:
The type of array will be inferred according to the return type of `ProductDto.covertToApi`


```suggestion
    return allProducts.map(ProductDto.covertToApi);
```